### PR TITLE
Cifar10 jax workload and reference submission

### PR DESCRIPTION
**Pull request by alpv95**
# Cifar-10 JAX

The cifar workload and reference submission are inspired by the existing imagenet and mnist implementations. 

Random crops and flips are used in the input pipeline, same as for imagenet.

The workload model is a ResNet-18 pulled from `imagenet_jax/models.py`. I wasn't sure about the target accuracy -- I can get to 85% in about 20 minutes with a single V100 or tesla P40 (all I have access to) for a single tuning run. I set it to 85% for now.

I use the same reference submission optimizer as imagenet, SGD with momentum + L2 and cosine decay, but with the specific tuning search space more optimized for cifar10 (according to e.g. https://arxiv.org/abs/1708.04552v2 ).

### CODE REVIEW

**Comment by fsschneider** [comment]
Hi Lawrence,
thanks for the great PR!

Could you take a look at the failed test and fix them?
Once this is done, we can give the PR a review and after that merge it :) 

@znado I assigned you as a reviewer, let me know if that doesn't work for you.

File: algorithmic_efficiency/workloads/cifar/cifar_jax/input_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+https://github.com/google/flax/blob/main/examples/imagenet/input_pipeline.py
+and adjusted to work for CIFAR10.
+"""
+
+from flax import jax_utils
+import jax
+import tensorflow as tf
+
+IMAGE_SIZE = 32
+RESIZE_SIZE = 40
```
Comment (znado) [inline algorithmic_efficiency/workloads/cifar/cifar_jax/input_pipeline.py:13]: is it standard to upscale the cifar images? I always used the original 32px ones.

File: algorithmic_efficiency/workloads/cifar/cifar_jax/workload.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  # parameters.
+  def is_output_params(self, param_key: spec.ParameterKey) -> bool:
+    pass
+
+  def initialized(self, key, model):
+    input_shape = (1, 32, 32, 3)
+    variables = jax.jit(model.init)({'params': key},
+                                    jnp.ones(input_shape, model.dtype))
+    model_state, params = variables.pop('params')
+    return params, model_state
```
Comment (znado) [inline algorithmic_efficiency/workloads/cifar/cifar_jax/workload.py:101]: nit: can we inline this in `init_model_fn` just so that people aren't tempted to call it?

**Review by znado — APPROVED** [review]
This is awesome, thank you so much for the contribution! Honestly not many comments to make, great job.

File: algorithmic_efficiency/workloads/cifar/cifar_jax/workload.py (hunk 2 above)
Comment (alpv95) [reply algorithmic_efficiency/workloads/cifar/cifar_jax/workload.py]: Done!

File: algorithmic_efficiency/workloads/cifar/cifar_jax/input_pipeline.py (hunk 1 above)
Comment (alpv95) [reply algorithmic_efficiency/workloads/cifar/cifar_jax/input_pipeline.py]: What I've mainly seen is upscaling by padding 4 on each side, then random crops, e.g. [pytorch example](https://pytorch-tutorial.readthedocs.io/en/latest/tutorial/chapter03_intermediate/3_2_2_cnn_resnet_cifar10/). 
However, it turns out _random_crop() here already does proper resizing of the crops, so I just removed the unnecessary RESIZE_SIZE and _center_crop().

**Comment by fsschneider** [comment]
@alpv95 Looks like there is one more linting issue and then this is ready to merge :tada:

Could you fix this and then I can merge it.

**Comment by alpv95** [comment]
> @alpv95 Looks like there is one more linting issue and then this is ready to merge 🎉
> 
> Could you fix this and then I can merge it.

Sorry, my auto-linters are a joke... all done.
